tests/__init__.py

```diff
@@ -42,22 +48,37 @@ def __init__(
             username,
             password,
             use_https,
+            access_tokens,
             debugmode,
             dsm_version,
         )
 
-    def _execute_get_url(self, request_url, append_sid=True):
+    def _execute_get_url(self, request_url):
         if VALID_DSM_HOST not in request_url or VALID_DSM_PORT not in request_url:
             return None
 
-        if (
-            self.LOGIN_URI in request_url
-            and VALID_USER in request_url
-            and VALID_PASSWORD in request_url
-        ):
-            return DSM_6_LOGIN
+        if self.LOGIN_URI in request_url:
```

Comment:
Add a test with user with 2SA that already grant this device with valid OTP code (so has a device token and no OTP code in the request).